### ISSUE TITLE
💅 Actually used private field considered unused (via `??=`)

### ISSUE THREAD

**Issue by jaens** [biomejs/biome#6994]
### Environment information

```bash
(see playground)
```

### Rule name

lint/correctness/noUnusedPrivateClassMembers

### Playground link

https://biomejs.dev/playground/?analyzerFixMode=safeAndUnsafeFixes&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABUAGUAcwB0ACAAewAKACAAIABwAHIAaQB2AGEAdABlACAAZgBpAGUAbABkADoAIABuAHUAbQBiAGUAcgA7AAoACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAHQAaABpAHMALgBmAGkAZQBsAGQAIAA%2FAD8APQAgADEAOwAKACAAIAB9AAoAfQA%3D&ruleDomains.react=recommended&ruleDomains.test=recommended&ruleDomains.next=recommended&ruleDomains.project=recommended

### Expected result

- `field` should not be considered unused here.
- Fixing should not delete the field.

```
export class Test {
  private field: number;

  constructor() {
    this.field ??= 1;
  }
}
```

(this is reduced, the original was more complicated, in particular there was a side effect on the right-hand of `??=` ...)

### Related

Looks like the private variable check has more issues: #6974, #6933

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by jaens** [comment]
I think some sort of random property-based / fuzz testing might have even caught this – with the property that running fixes must not produce code with type errors from code that had none.

**Comment by vladimir-ivanov** [comment]
https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4BVhvgbwCg44woBLANxRmDgDNzgkATALjgDsBXAWwCNgUANyFiiCJzxRuCGNAAUASjhESJGAAtyaAHSNmLOAH5jAXjgBGUSQC%2BhW0A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false please see here, this is same behaviour as eslint.

Please note that this.someProperty in constructor behaves differently from other class methods.

e.g. this one is not flagged:
https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4BVhvgbwCg44woBLANxRmDgDNzgkATALjgDsBXAWwCNgUANyFicNBF7AAssBgALCCwAUASjhESJReTQA6RsxZwA-KYC8cAIyiSAX0KOgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

That said:

https://biomejs.dev/playground/?analyzerFixMode=safeAndUnsafeFixes&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABUAGUAcwB0ACAAewAKACAAIABwAHIAaQB2AGEAdABlACAAZgBpAGUAbABkADoAIABuAHUAbQBiAGUAcgA7AAoACgAgACAAcwBvAG0AZQBNAGUAdABoAG8AZAAoACkAIAB7AAoAIAAgACAAIAB0AGgAaQBzAC4AZgBpAGUAbABkACAAPwA%2FAD0AIAAxADsACgAgACAAfQAKAH0ACgA%3D&ruleDomains.react=recommended&ruleDomains.test=recommended&ruleDomains.next=recommended&ruleDomains.project=recommended this one should not be flagged. 

Will address. Thanks for raising

**Comment by vladimir-ivanov** [comment]
this has been already fixed, but not deployed yet. Will however write a test for it, to avoid future regressions.

**Comment by vladimir-ivanov** [comment]
https://github.com/biomejs/biome/issues/6933 - this is a different rule failure 

lint/correctness/noUnusedPrivateClassMembers: 
This private class member is defined but never used.

will look into that separately. Thank you
